deep-learning/_posts/2017-03-21-xception.md

```diff
@@ -0,0 +1,66 @@
+
+There is two minor differences between a depthwise separable convolution and the module shown in figure 4 : the order of the two operations, and the presence or absence of a non-linearity between those operations. The author argues that the first does not matter. The second difference is important, and **the absence of non-linearity improves accuracy** (see figure 10). An opposite result is reported by Szegedy et al. for Inception modules. Chollet provides an explanation in section 4.7.
+
+![Figure 10](/deep-learning/images/xception/fig10.png)
+
+### The Xception Architecture
+
+In short, the Xception architecture is a linear stack of depthwise separable convolution layers with residual connections.
+
+![Xception architecture](/deep-learning/images/xception/architecture.png)
+
+### Experimental evaluation
+
+Two datasets were used : ImageNet (1000 classes) and JFT, an internal Google dataset comprising 350 million high-res images with labels from a set of 17,000 classes. For each dataset, an Inception V3 model and an Xception model were trained. Models trained on JFT were evaluated using a smaller _auxiliary_ dataset, named FastEval14k. The networks were trained on 60 NVIDIA K80 GPUs each[^2]. The ImageNet experiments took 3 days each, while the JFT experiments took over **one month each**. Note the great amount of resources invested, both in terms of money and time.
+
+On ImageNet, Xception shows marginally better results than Inception V3. On JFT, Xception shows a **4.3% relative improvement on the FastEval14k MAP@100 metric** (see table 1). The fact that both architectures have almost the same number of parameters indicates that the improvement seen on ImageNet and JFT does not come from added capacity but rather from a **more efficient use of the model parameters**. The author also point out that Xception outperforms ImageNet results reported by He et al. for ResNet-50, ResNet-101 and ResNet-152.
+
+**Table 1**: Classification performance comparison on JFT (single crop, single model). The models with FC layers have two 4096-unit FC layers before the logistic regression layer at the top. MAP@100 is the mean average precision on the top 100 guesses.
+
+|-------------------------------+-------------------------|
+| Model                         | JFT FastEval14k MAP@100 |
+|:------------------------------|:-----------------------:|
+| Inception V3 - no FC layers   | 6.36                    |
+| Inception V3 - with FC layers | 6.50                    |
+| Xception - no FC layers       | 6.70                    |
+| Xception - with FC layers     | **6.78**                |
+|-------------------------------+-------------------------|
+
+<p><!-- spacing --></p>
+
+### Future directions
+
+The author says that there is no reason to believe that depthwise separable convolutions are optimal. It may be that **intermediate points on the spectrum, lying between regular Inception modules and depthwise separable convolutions, hold further advantages**.
```

Comment:
@jodoin I think I will try an hyper-parameter search on that spectrum for Mio-TCD!